simtools/model/camera.py

```
@@ -0,0 +1,670 @@
+
+        Parameters
+        ----------
+        cameraConfigFile: string
+            The sim_telarray file name.
+
+        Returns
+        -------
+        dict: pixels
+            A dictionary with the pixel positions, the camera rotation angle,
+            the pixel shape, the pixel diameter, the pixel IDs and their "on" status.
+
+        Notes
+        -----
+        The pixel shape can be hexagonal (denoted as 1 or 3) or a square (denoted as 2).
+        The hexagonal shapes differ in their orientation, where those denoted as 3 are rotated
+        clockwise by 30 degrees with respect to those denoted as 1.
+        '''
+
+        datFile = open(cameraConfigFile, 'r')
+        pixels = dict()
+        pixels['diameter'] = 9999
+        pixels['funnelShape'] = 9999
+        pixels['rotateAngle'] = 0  # The LST and MST-NectarCam cameras need to be rotated
+        pixels['x'] = list()
+        pixels['y'] = list()
+        pixels['pixID'] = list()
+        pixels['pixOn'] = list()
+        for line in datFile:
+            if line.startswith('PixType'):
+                pixels['funnelShape'] = int(line.split()[5].strip())
+                pixels['diameter'] = float(line.split()[6].strip())
+            if line.startswith('Rotate'):
+                pixels['rotateAngle'] = np.deg2rad(float(line.split()[1].strip()))
+            if line.startswith('Pixel'):
+                pixInfo = line.split()
+                pixels['x'].append(float(pixInfo[3].strip()))
+                pixels['y'].append(float(pixInfo[4].strip()))
+                pixels['pixID'].append(int(pixInfo[1].strip()))
+                if len(pixInfo) > 9:
+                    if int(pixInfo[9].strip()) != 0:
+                        pixels['pixOn'].append(True)
+                    else:
+                        pixels['pixOn'].append(False)
+                else:
+                    pixels['pixOn'].append(True)
+
+        if pixels['diameter'] == 9999:
```

Maybe we could add an if to verify the funnelShape as well. I think the code will break in some other parts if funnelShape not 1, 2 or 3.

---

Sure, added one and also raised a ValueError instead of defaulting to some random value for the pixel diameter.